packages/eas-json/src/utils.ts

```diff
@@ -30,21 +30,33 @@ export class EasJsonUtils {
 
   public static getBuildProfileDeprecationWarnings(
     buildProfile: BuildProfile,
-    profileName?: string
+    profileName?: string,
+    rawEasJson?: any
   ): EasJsonDeprecationWarning[] {
     const warnings: EasJsonDeprecationWarning[] = [];
+    const finalProfileName = profileName ?? 'production';
 
     if (buildProfile.cache?.cacheDefaultPaths !== undefined) {
       warnings.push({
         message: [
-          `The "build.${
-            profileName ?? 'production'
-          }.cache.cacheDefaultPaths" field in eas.json is deprecated and will be removed in the future.`,
+          `The "build.${finalProfileName}.cache.cacheDefaultPaths" field in eas.json is deprecated and will be removed in the future.`,
         ],
         docsUrl: 'https://docs.expo.dev/build-reference/caching/#ios-dependencies',
       });
     }
 
+    if (
+      buildProfile.cache?.customPaths !== undefined ||
+      rawEasJson?.build?.[finalProfileName]?.cache?.customPaths !== undefined
```

Comment:
It's a bit problematic here, since the raw JSON doesn't have the merged profiles as the buildProfile does. Also, the previous method implemented above also doesn't check for that 🤔 
What would be the best approach here? Looping through all the profiles (regardless of their name) and checking all of them?

Reply:
I've changed the singular check into a recursive function going deeper through the `extends` key until the extensions finish or a deprecated field is found